datalad/support/gitrepo.py

```diff
@@ -3424,16 +3429,35 @@ def save_(self, message=None, paths=None, _status=None, **kwargs):
             if (props.get('state', None) in ('modified', 'untracked') and
                 not (f in to_add_submodules or f in to_stage_submodules))}
         if to_add:
+            if self.config.obtain("datalad.save.windows-compat-warning") in \
+                    ('warning', 'error'):
+                # check that non-Windows users generate win-compatible filenames
+                to_add, problems = self._check_for_win_compat(to_add)
             lgr.debug(
                 '%i path(s) to add to %s %s',
                 len(to_add), self, to_add if len(to_add) < 10 else '')
-            for r in self._save_add(
-                    to_add,
-                    git_opts=None,
-                    **{k: kwargs[k] for k in kwargs
-                       if k in (('git',) if hasattr(self, 'annexstatus')
-                                else tuple())}):
-                yield r
+            if to_add:
+                for r in self._save_add(
+                        to_add,
+                        git_opts=None,
+                        **{k: kwargs[k] for k in kwargs
+                           if k in (('git',) if hasattr(self, 'annexstatus')
+                                    else tuple())}):
+                    yield r
```

Comment:
a minor pythonisation:

```suggestion
                yield from self._save_add(
                        to_add,
                        git_opts=None,
                        **{k: kwargs[k] for k in kwargs
                           if k in (('git',) if hasattr(self, 'annexstatus')
                                    else tuple())}):
```
? since _save_add  is a generator